src/orderbook_fetcher.ts

```diff
@@ -18,13 +19,23 @@ export class OrderbookFetcher {
 
   constructor(readonly web3: Web3, pageSize: number, pollFrequency: number, logger: CategoryLogger) {
     const poll = async () => {
-      try {
-        const contract = await this.loadBatchExchangeViewer()
-        this.orderbooks = await updateOrderbooks(contract, pageSize, logger)
-      } catch (error) {
-        logger.error(`Failed to fetch Orderbooks: ${error}`, null)
-      }
-      this.intervalId = setTimeout(poll, pollFrequency)
+      const { totalCount, errorsCount, durationsTotals } = orderBookFetcher
+
+      executeWithMetrics({
+        totalCount,
+        errorsCount,
+        durationsTotals,
+
+        runnable: async () => {
+          try {
+            const contract = await this.loadBatchExchangeViewer()
+            this.orderbooks = await updateOrderbooks(contract, pageSize, logger)
+          } catch (error) {
+            logger.error(`Failed to fetch Orderbooks: ${error}`, null)
+          }
+          this.intervalId = setTimeout(poll, pollFrequency)
```

Comment:
Since this is now a timeout, not interval, maybe rename, and use `clearTimeout` in `terminate()`
Though as you know, both work :wink:

Reply:
Heheh

Yep. Indeed they work, and indeed we should make it match it for readability. Good eye!
That's not from this PR, but I'll do it here anyways